### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by toddmotto**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,88 @@
...
+
+Tranditionally, server-side rendering would be like the following -
+
+![Server-side rendering](http://www.michaelgallego.fr/images/posts/2012-11-26-client-side-1.png)
+
+With Angular, the flow will go like this instead -
+
+![Client-side rendering](http://branchandbound.net/pics/client-side-rendering-only.png)
+
+(this looks a lot more complicated, but it isn't!)
```
Comment (ipc103) [inline README.md]: The tone here is nice, but you don't tell the reader why it isn't complicated! Help the reader understand why it's not complicated. Break down the concept with a step-by-step explanation.

File: README.md [hunk 2]
```diff
@@ -1 +1,88 @@
...
+
+This is extremely powerful - it means that we can inject our custom Controllers etc. into all of our code, and have every dependency available in the application.
+
+## Single Page Applications
+
+Angular allows us to create single page applications (SPAs). The means that the user goes to one page and very rarely will have to refresh/leave that page to go to other parts of the website.
+
+Single page applications handle all the routing, calling different parts of your codebase dependant on what URL the user is currently at.
+
+Tranditionally, server-side rendering would be like the following -
```
Comment (ipc103) [inline README.md]: These graphics are really confusing - don't include them without a step-by-step explanation of what they represent.

File: README.md [hunk 3]
```diff
@@ -1 +1,88 @@
...
+
+![Two-way data-binding](https://docs.angularjs.org/img/Two_Way_Data_Binding.png)
+
+## Dependency Injection
+
+Angular offers dependency injection. This allows us to request whatever Controller/Directive/etc. we want, instead of receiving them all in a specific order. This is not a default JavaScript behaviour. 
+
+This is extremely powerful - it means that we can inject our custom Controllers etc. into all of our code, and have every dependency available in the application.
+
+## Single Page Applications
```
Comment (ipc103) [inline README.md]: This could probably come first in the README - what is a single page application and why is it important? Once the reader understands that, they'll feel much more invested in learning about the pieces of Angular. How do the other pieces of Angular break down to help us build Single Page Applications?

File: README.md [hunk 4]
```diff
@@ -1 +1,88 @@
-# angular-concepts-readme
+# Concepts
```
Comment (ipc103) [inline README.md]: Give this a more descriptive title - 'Angular Concepts' or 'Angular Core Concepts' would be better

File: README.md [hunk 5]
```diff
@@ -1 +1,88 @@
...
+
+## Overview
+
+There are some core concepts that every Angular developer will need to familiarise themselves with. This will teach you the core concepts that you will be using in every Angular application you make.
```
Comment (ipc103) [inline README.md]: Try to begin with a friendlier introduction to draw the reader in here. "In this lesson, we'll be looking at some of the core concepts of Angular and how they differ from Rails. Some will be familiar to you and some will be different - don't worry if it takes some time to sink in! "

File: README.md [hunk 6]
```diff
@@ -1 +1,88 @@
...
+
+## Objectives
+
+- Define MVC, MVVM
```
Comment (ipc103) [inline README.md]: spell out fully on first use - "Model-View-Controller (MVC) and Model-View-ViewModel (MVVM).

File: README.md [hunk 7]
```diff
@@ -1 +1,88 @@
...
+- Describe two-way data-binding
+- Demonstrate updating Models and Views without two-way binding
+- Describe Angular's two-way binding with flow charts
+- Describe Dependency Injection
```
Comment (ipc103) [inline README.md]: try to be more specific  with your objectives. It's hard to know as a reader if I'm able to describe a topic or not.

File: README.md [hunk 8]
```diff
@@ -1 +1,88 @@
...
+- Describe Angular's Dependency Injection
+- Describe a Single Page Application and common features
+- Demonstrate server-side rendering concepts
+- Describe Single Page Applications with Angular
+
+## MVC/MVVM
+
+You might have seen the term MVC/MVVM being thrown around. These stand for Model-View-Controller/Model-View-ViewModel respectively. These is how you architecture your software. They are both quite similar, however there are a few differences -
```
Comment (ipc103) [inline README.md]: MVC is a nice place to start here - the students will be familiar with this from previous lessons.  Still, it's nice to give a refresher. Remind them why we architecture our software this way. 

Keep looking for ways to connect and reinforce things that students already know.

File: README.md [hunk 9]
```diff
@@ -1 +1,88 @@
...
+
+### MVC (Model-View-Controller)
```
Comment (ipc103) [inline README.md:45]: Feel free to use an analogy - we'll often say that MVC is like a restaurant - the model is the chef who prepares the food, or data, the view is the table, or presentation, and the controller is the waiter, passing the data, or food, between the two.

Again, this is a great place to start - start where we know, then present a new concept. This might seem simple, but any place that you can connect two things clearly is gold. Remind us about what MVC is, and then explain MVVM in that context.

File: README.md [hunk 10]
```diff
@@ -1 +1,88 @@
...
+
+The view is the user interface. This takes data from the model and displays it nicely for the user. Any user interaction in the view is then handed over to the controller.
+
+The controller receives user input and makes calls to the model and view to update/modify either. This keeps the main aspects of our application separate.
+
+### MVVM (Model-View-ViewModel)
+
+Similar to MVC, however the controller is replaced with a ViewModel.
+
+The ViewModel is responsible for all view logic - changing the model data appropriately for use in the view. (For example, our model might contain a unix timestamp for a date but we need it in human readable format). Your model will contain all of your business logic, and your view is there to display your model (modified by the ViewModel).
```
Comment (ipc103) [inline README.md]: Try to explain what a ViewModel is in terms of a controller. How are they similar, and how are they different? Again, use terms that the reader will already know. Is there an analogy you can use? Can you somehow fit it into the restaurant analogy?

File: README.md [hunk 11]
```diff
@@ -1 +1,88 @@
...
+
+MVVM is the preferred architecture when developing Angular applications, and it is what we will be using.
```
Comment (ipc103) [inline README.md]: Why is MVVM preferred? What are the alternatives? Avoid statements like this unless you can take the time to explain them. If it's not important to understand at this point, you can leave it out.

File: README.md [hunk 12]
```diff
@@ -1 +1,88 @@
...
+
+## Data-binding
+
+Data-binding is the term for automatic synchronisation of data between the model and the view. This comes in two formats - one-way and two-way data-binding.
```
Comment (ipc103) [inline README.md]: Make this explanation more accessible. Use an analogy or a real-world example in terms of what the user already knows.

File: README.md [hunk 13]
```diff
@@ -1 +1,88 @@
...
+
+### MVC (Model-View-Controller)
+
+Three parts to MVC; model, view, controller. The model manages the fundamental behaviours and data of the application. This could be retrieved from a database or any other sort of data store.
+
+The view is the user interface. This takes data from the model and displays it nicely for the user. Any user interaction in the view is then handed over to the controller.
+
+The controller receives user input and makes calls to the model and view to update/modify either. This keeps the main aspects of our application separate.
+
+### MVVM (Model-View-ViewModel)
```
Comment (ipc103) [inline README.md:61]: You don't mentioned the MV of MVVM in this section - this becomes confusing later when you start referencing models. Are models in Angular the same as models in Rails? When you say 'model', are you referring to the 'model' of a Rails app? Be accessible but clear: "In a Rails application, our models represent data in our database. In Angular..."

File: README.md [hunk 14]
```diff
@@ -1 +1,88 @@
...
+
+## Data-binding
+
+Data-binding is the term for automatic synchronisation of data between the model and the view. This comes in two formats - one-way and two-way data-binding.
+
+### One-way data-binding
+
+Imagine you have a form input, asking for the user's name. You've also got a `<h2>` tag below, that you want to populate to say "Hey, {name}!". Wouldn't it be neat if this `<h2>` tag could update as the user types in their name? This is called one-way data-binding. The model (our user's name) is updated and our view is updated to reflect the change to our model.
+
+Why isn't this two-way? Well, the `<h2>` tag populated with the user's name (model) will never update the model itself. Therefore, the flow of data goes from the input to the `<h2>` tag, but not from the `<h2>` tag to the input.
```
Comment (ipc103) [inline README.md]: Here, you're explaining why this isn't two-way data binding without explaining what two-way data-binding is. This is confusing - just explain what two-way data-binding is, and the reader will understand the difference.

File: README.md [hunk 15]
```diff
@@ -1 +1,88 @@
...
+
+MVVM is the preferred architecture when developing Angular applications, and it is what we will be using.
+
+## Data-binding
+
+Data-binding is the term for automatic synchronisation of data between the model and the view. This comes in two formats - one-way and two-way data-binding.
+
+### One-way data-binding
+
+Imagine you have a form input, asking for the user's name. You've also got a `<h2>` tag below, that you want to populate to say "Hey, {name}!". Wouldn't it be neat if this `<h2>` tag could update as the user types in their name? This is called one-way data-binding. The model (our user's name) is updated and our view is updated to reflect the change to our model.
```
Comment (ipc103) [inline README.md]: "Wouldn't it be neat if this `<h2>` tag could update as the user types in their name?" - good tone here. More of this!

Build it out a little bit more, give more context to this example. Give a simple example of a site where you might see something like this.

File: README.md [hunk 16]
```diff
@@ -1 +1,88 @@
...
+
+Why isn't this two-way? Well, the `<h2>` tag populated with the user's name (model) will never update the model itself. Therefore, the flow of data goes from the input to the `<h2>` tag, but not from the `<h2>` tag to the input.
+
+[Here's an example of one-way data-binding.](https://jsfiddle.net/6z6d5b7x/)
```
Comment (ipc103) [inline README.md]: This code sample is too long to be presented in one chunk - the student will have a difficult time parsing any meaning out of it. If the goal is to demo just the functionality, have them build it out themselves. Building this out from scratch could probably be it's own lesson. 

In general, if you want to include a code sample, include it directly in the README. This will be difficult to maintain and could pull the student out of the flow of the lesson.

File: README.md [hunk 17]
```diff
@@ -1 +1,88 @@
...
+
+Why isn't this two-way? Well, the `<h2>` tag populated with the user's name (model) will never update the model itself. Therefore, the flow of data goes from the input to the `<h2>` tag, but not from the `<h2>` tag to the input.
+
+[Here's an example of one-way data-binding.](https://jsfiddle.net/6z6d5b7x/) 
+
+### Two-way data-binding 
+
+However, you have two-way data-binding when you bind your model value to an element that can both change and display the value of the model (such as an input).
+
+You aren't restricted to updating your model value just in the `<input />` however - you can update this model value elsewhere (such as a "reset" button to reset the model) and it will be automatically reflected in the input (it will go blank).
```
Comment (ipc103) [inline README.md]: Here, start with explaining the behavior/feature you're trying to implement first.  "But what if we want our model to be changed by something in addition to the input field, such as a 'reset' button? Wouldn't it be nice if we could press 'reset' and have that clear both the h2 and our input?..."

File: README.md [hunk 18]
```diff
@@ -1 +1,88 @@
...
+
+[Checkout this example of two-way data-binding](https://jsfiddle.net/6z6d5b7x/1/)
```
Comment (ipc103) [inline README.md:95]: See the note above about these code samples

**Comment by ipc103** [comment]
There are too many different concepts presented at once here without context. By the time I need to use dependency injection in my Application, I'll have forgotten all about it. Try not to introduce the solution to a problem without demonstrating what the problem is first. For example, our Sinatra unit begins with having the user create an application in a single file, and new concepts and features are introduced one at a time. At what point does it make sense to introduce the student to each of these different concepts?

File: README.md [hunk 19]
```diff
@@ -1 +1,109 @@
-# angular-concepts-readme
+# Angular Core Concepts
```
Comment (ipc103) [inline README.md]: Based on the content here, 'MVVM and Data Binding in Angular' would be a better title

File: README.md [hunk 20]
```diff
@@ -1 +1,109 @@
...
+
+## Overview
+
+In this lesson, we are going to look at the core concepts that we will use in every Angular application. Some will be familiar to you, however some may seem a bit strange. Don't worry if they take time to sink in!
```
Comment (ipc103) [inline README.md:5]: Great intro! This sets the tone really well. Feels very reassuring and human.

File: README.md [hunk 21]
```diff
@@ -1 +1,109 @@
...
+
+## Objectives
```
Comment (ipc103) [inline README.md]: Give a little bit more context around the objectives - for example, "Define MVC (Model-View-Controller) and MVVM (Model-View-ViewModel) as they relate to single page web applications". Without a conditional or context, the objectives can be interpreted very broadly and the student can get confused.

File: README.md [hunk 22]
```diff
@@ -1 +1,109 @@
...
+- Define MVC (Model-View-Controller), MVVM (Model-View-ViewModel)
+- Describe one-way and two-way data-binding
+- Describe Angular's two-way binding
+- Describe Dependency Injection
+- Describe a Single Page Application and common features
+- Demonstrate server-side rendering concepts
+
+## Single Page Applications
+
+Angular allows us to create single page applications (SPAs). The means that the user goes to one page and very rarely will have to refresh/leave that page to go to other parts of the website.
```
Comment (ipc103) [inline README.md]: Similar to the previous README, can you start with a few examples of SPAs that the reader might be familiar with? "What do web applications like x, y, and z have in common? Have you noticed that when you go to one page you very rarely will have to refresh/leave that page to go to other parts of the website? These are what we call single page applications." This will let the reader instantly make a connection to something they've seen before.

File: README.md [hunk 23]
```diff
@@ -1 +1,109 @@
...
+
+Single page applications handle all the routing, calling different parts of your codebase dependant on what URL the user is currently at.
```
Comment (ipc103) [inline README.md]: Can you flesh this out more, maybe with an example of a real application? Doesn't have to be technical. As you discuss more about how the routing works, this example may also be a good transition into MVC/MVVM - move it below the description of traditional rendering below

File: README.md [hunk 24]
```diff
@@ -1 +1,109 @@
...
+
+Tranditionally, server-side rendering would be like the following -
+
+![Server-side rendering](http://www.michaelgallego.fr/images/posts/2012-11-26-client-side-1.png)
+
+The user requests the page from the server, and the server will return all the HTML to display on that page. The data would not be updated until the user refreshes. This puts overhead on the server, as it will have to render the page on every page load.
```
Comment (ipc103) [inline README.md]: Nice explanation of server side rendering here! Can you incorporate the example below somehow? IE if 1000 users go to this website, the server is rendering 1000 different finished HTML templates.

File: README.md [hunk 25]
```diff
@@ -1 +1,109 @@
...
+
+With Angular, the flow will go like this instead -
+
+![Client-side rendering](http://branchandbound.net/pics/client-side-rendering-only.png)
+
+This looks a lot more complicated, but it isn't! Instead of returning rendered HTML for each user, the server would return the same page for every user. The client will then make a request to the server, requesting what's necessary for that page to render (such as a list of products). The client will then render this page for the user.
+
+This has massive advantages - if 1000 users are on the website, instead of one server rendering 1000 pages, 1000 clients are rendering one page, reducing strain on the server. We can also re-use backend endpoints - our mobile website and our desktop website can both request the same data from the server but display it differently.
```
Comment (ipc103) [inline README.md]: This makes much more sense now! Can you break it down even further? Make the example more specific - name an app where the server would send back a completely rendered template vs one where the server sends a blank template w/ data and the client renders it. Using a specific example that the reader is familiar with will make this super clear.

File: README.md [hunk 26]
```diff
@@ -1 +1,109 @@
...
+
+## MVC/MVVM
+
+You might have seen the term MVC/MVVM being thrown around. These stand for Model-View-Controller/Model-View-ViewModel respectively.
```
Comment (ipc103) [inline README.md]: This is nit-picky, but use "and" instead of "/" - it will read better.

File: README.md [hunk 27]
```diff
@@ -1 +1,109 @@
...
+
+This looks a lot more complicated, but it isn't! Instead of returning rendered HTML for each user, the server would return the same page for every user. The client will then make a request to the server, requesting what's necessary for that page to render (such as a list of products). The client will then render this page for the user.
+
+This has massive advantages - if 1000 users are on the website, instead of one server rendering 1000 pages, 1000 clients are rendering one page, reducing strain on the server. We can also re-use backend endpoints - our mobile website and our desktop website can both request the same data from the server but display it differently.
+
+## MVC/MVVM
+
+You might have seen the term MVC/MVVM being thrown around. These stand for Model-View-Controller/Model-View-ViewModel respectively. 
+
+These are concepts for how we architecture our software. We do this to promote separation of concerns. Separation of concerns is dividing our code into logical chunks, with each chunk addressing a separate concern. This allows us to create testable, reusable and scalable well designed code.
```
Comment (ipc103) [inline README.md:41]: Great paragraph - this is something that comes up a lot in the curriculum and this is a perfect place to reemphasize it. Never be afraid to repeat important concepts. It's really helpful for students to see stuff again.

File: README.md [hunk 28]
```diff
@@ -1 +1,109 @@
...
+
+### MVC (Model-View-Controller)
+
+Three parts to MVC; model, view, controller. The model manages the fundamental behaviours and data of the application. This could be retrieved from a database or any other sort of data store.
+
+The view is the user interface. This takes data from the model and displays it nicely for the user. Any user interaction in the view is then handed over to the controller.
+
+The controller receives user input and makes calls to the model and view to update/modify either. This keeps the main aspects of our application separate.
+
+As easy way to visualize this is if we imagine we are ordering a coffee. The Model is the information about the drink we've ordered, the Controller is the barista and the View is the finished drink. The barista would make the drink from our order (the Model), and serve it to us (the View).
```
Comment (ipc103) [inline README.md]: First of all, this is a great example - really easy to visualize what's happening here. One issue that students may have is this concept feels slightly different from the way MVC works in Rails. Here's the analogy we use from the [Rails Model View Controller Readme](
https://github.com/learn-co-curriculum/rails-model-view-controller-readme/)

> In a restaurant kitchen we have three key components:

> A chef that makes the food

> A waiter that takes the order and brings out the food

> A table where the food is served to the customer

> Using this as an analogy, we can assign the following roles to each of our restaurant components:

> Model - The model is the chef, it should manage the critical aspects of the application. One of my favorite tasks in a Rails application is working with the model files. This is where you can be very expressive with the custom algorithms that you want to utilize and you also have direct access to the specific database record. The logic of your application should mainly reside in the model files.

> Controller - The controller is the waiter. In the same way that the waiter takes the order from the customer to the chef and then brings the food to the table, the controller transmits data requests from the user to the model, and then delivers data that is rendered in the view to the user.

> View - The view is the table. A table shouldn't do anything besides sit there and hold the food when it is delivered. In like manner the views should not contain any programming logic, they should simply render what the controller sends it. One of the top issues I discover when I review a Rails application that has a large number of bugs is that the developer integrated too much logic directly into the view.

Wouldn't the Model in MVC, or at least in MVC in Rails, be the barista? The information about the drink is the data. The model prepares the data (coffee) and the barista taking the order passes it from the barista to the customer. 

Generally, this feels very different than the way we talk about it in Rails, so make sure to break it down and be super clear. If MVC is different in the context of front-end frameworks, that's okay, just directly say "MVC in this context is different than what you've seen before."

File: README.md [hunk 29]
```diff
@@ -1 +1,109 @@
...
+
+So for example, if you walked into a coffee bar and ordered a hot chocolate, you'd get a hot chocolate. If you then wanted marshmallows, you'd ask the barista, who would go back to the till and modify your order to add on marshmallows. They would also then put marshmallows in your drink, and return you the updated drink (the View).
```
Comment (ipc103) [inline README.md:55]: Having said all that, love that you come back to the analogy here. With traditional rendering, does the barista have to remake you an entire new drink when you ask for one with marshmallows? If so, that might be a fun thing to include here, as it would really hammer home why MVVM is important.

File: README.md [hunk 30]
```diff
@@ -1 +1,109 @@
...
+
+### MVVM (Model-View-ViewModel)
+
+Similar to MVC, however the controller is replaced with a ViewModel.
+
+The ViewModel is responsible for all view logic - changing the model data appropriately for use in the view. (For example, our model might contain a unix timestamp for a date but we need it in human readable format). Your model will contain all of your business logic, and your view is there to display your model (modified by the ViewModel).
```
Comment (ipc103) [inline README.md]: Move this paragraph to directly after you finish your coffee shop example. This will make more sense after seeing the analogy.

File: README.md [hunk 31]
```diff
@@ -1 +1,109 @@
...
+
+Let's take a trip back to the coffee shop. You'll notice there is a new member of staff - a cashier. The barista was too overloaded as a controller, and has stepped down to just being a helper. The cashier has been brought in as a ViewModel, in order to make the coffee shop more efficient.
```
Comment (ipc103) [inline README.md:65]: "Let's take a trip back to the coffee shop" - really great tone there!

File: README.md [hunk 32]
```diff
@@ -1 +1,109 @@
...
+
+Like before, we order a hot chocolate. The cashier takes your order, updating the Model. They then shout to the barista that they need a hot chocolate. The barista makes the hot chocolate, and returns it to the cashier. The cashier then writes your name on the lid, and returns it to you.
+
+The differences here is that we no longer have one person taking your order and delivering your drink. Instead, the cashier (ViewModel) uses a helper (business logic) to get your drink order (Model) completed and only modifies the lid with your name (view logic) and returns it to you (the View).
```
Comment (ipc103) [inline README.md:69]: So, the difference is that the cashier now has some control of the data? Not every request has to go through to the barista? Is that it? If so, then I get it!

In future READMEs, definitely come back to this example. I think I have a good understanding of this concept, but I want to be able to test it as we get more technical in the future. Leave a reassuring note here for reader letting them know that we'll take a more technical look at this example in future lessons.

File: README.md [hunk 33]
```diff
@@ -1 +1,109 @@
...
+
+This is the preferred architecture for Angular applications. Why? We no longer have a controller doing everything for us. Instead, we can simplify the application down massively by having helpers that we can use everywhere.
```
Comment (ipc103) [inline README.md:71]: Another great line - this is a great close for this section and presents the solution really well after seeing the problem. We don't want to overwhelm the barista, so we use this pattern. Great conclusion to this section

File: README.md [hunk 34]
```diff
@@ -1 +1,109 @@
...
+
+## Data-binding
```
Comment (ipc103) [inline README.md:75]: We're almost there, but this transition feels jarring. Can you begin this section with the coffee shop example again? How does data-binding tie in? Would this be communication between the cashier and the barista? 

Ask more questions - we're getting a good sense of this coffee shop, and we're seeing the problems that MVVM can solve. How do they keep track of the order? This feels like the problem that data binding can solve.

File: README.md [hunk 35]
```diff
@@ -1 +1,109 @@
...
+
+Data-binding is the term for automatic synchronisation of data between the model and the view. This comes in two formats - one-way and two-way data-binding.
```
Comment (ipc103) [inline README.md]: this is too technical for an intro - save descriptions like for after you've used an analogy

File: README.md [hunk 36]
```diff
@@ -1 +1,109 @@
...
+
+### One-way data-binding
```
Comment (ipc103) [inline README.md:79]: Can you explain these in terms of the coffee shop first, then go into your awesome second example of the search Query? The more examples/analogies here, the better.

File: README.md [hunk 37]
```diff
@@ -1 +1,109 @@
...
+
+Imagine you have a form input, asking for the user's name. You've also got a `<h2>` tag below, that you want to populate to say "Hey, {name}!". Wouldn't it be neat if this `<h2>` tag could update as the user types in their name? This is called one-way data-binding. The model (our user's name) is updated and our view is updated to reflect the change to our model.
+
+For example, you'd use one-way data-binding when you've got a search box - the user can type in their search query and the title can change to "Results for: {searchQuery}" - pretty cool!
+
+[Here's an example of one-way data-binding.](https://jsfiddle.net/6z6d5b7x/) 
+
+### Two-way data-binding
```
Comment (ipc103) [inline README.md:89]: Same as above, can you start with the coffee shop? Not sure what the 'reset' button would be - maybe two people are placing one order and they both can modify it? Going into the below example after the coffee shop will make it very clear

File: README.md [hunk 38]
```diff
@@ -1 +1,109 @@
...
+
+You can see the flow of this here -
+
+![Two-way data-binding](https://docs.angularjs.org/img/Two_Way_Data_Binding.png)
+
+Two-way data-binding has a one-directional flow of data which ensures that the View and the Model data is always synchronized.
+
+In the above example our template compiles to become our View, which is a representation of the Model data. The View can then change our Model (for instance, an `<input />` being updated), which then updates our View. Model changes drive View changes and View changes drive Model changes. This keeps everything always synchronized. 
+
+## Dependency Injection
```
Comment (ipc103) [inline README.md]: This feels like it should be in a different README later on - we don't need to know about Dependency Injection yet, let this one just be about MVVM and data binding, this feels like too much of a left turn

**Comment by ipc103** [comment]
Overall, really excellent tone here - nice work! I'm really enjoying the coffee shop example, I think will be super clear for the readers. Again, just some questions about flow - how can you carry that analogy into the section about data binding. 

Along that lines, the Dependency Injection section feels shoehorned in at the end. The README works well without it and it doesn't seem important to understanding the other concepts presented. In general, don't bother bringing up concepts until you have to. By the time we go to use dependency injection, students will have forgotten that they read that paragraph. 

Overall though, awesome improvements, thanks very much!

File: README.md [hunk 39]
```diff
@@ -1 +1,110 @@
...
+
+With Angular, the flow will go like this instead -
+
+![Client-side rendering](http://branchandbound.net/pics/client-side-rendering-only.png)
+
+This looks a lot more complicated, but it isn't! Instead of returning rendered HTML for each user, the server would return the same page for every user. The client will then make a request to the server, requesting what's necessary for that page to render (such as a list of products). The client will then render this page for the user.
+
+This has massive advantages - if 1000 users are on the website, instead of one server rendering 1000 pages, 1000 clients are rendering one page, reducing strain on the server. We can also re-use backend endpoints - our mobile website and our desktop website can both request the same data from the server but display it differently.
+
+One example of this that you can checkout for yourselves is if you go to Facebook. You might notice how you don't actually see anything until the content very quickly and then the client renders this - saving so many resources in the mean time!
```
Comment (ipc103) [inline README.md]: Great example, but the wording here is a little bit strange. Can you rephrase this?

File: README.md (hunk 29 above)
Comment (AnnJohn) [inline README.md:55]: resolved

File: README.md (hunk 32 above)
Comment (AnnJohn) [inline README.md:69]: resolved

File: README.md (hunk 33 above)
Comment (AnnJohn) [inline README.md:71]: resolved
